client/collection.go

```diff
@@ -16,24 +16,34 @@ import (
 	"github.com/sourcenetwork/defradb/datastore"
 )
 
+// CollectionDefinition contains the metadata defining what a Collection is.
+type CollectionDefinition struct {
```

Comment:
thought: I'm a little worried that the name is so close to `CollectionDescription` that it will be a source of confusion. What do you think of `CollectionInfo`?

Reply:
I dislike `CollectionInfo` as it suffers from being a little ambiguous - could be read as containing info about what data is in the collection too, `CollectionDefinition` suffers less from that.

Reply:
We should probably think of something better then.

Reply:
> We should probably think of something better then.

I tried, and `CollectionDefinition` was the best I could come up with without spending an unreasonable amount of time on it.  I'm fairly happy with it given the problem.

Reply:
Just noting here for future reference:

I feel like `client.SchemaDescription` should just be `client.Schema` and `client.CollectionDescription` should be `client.Collection`. The interface should have a different name. Probably something like `client.CollectionAPI` or `client.CollectionHandler`. 

This way, `client.CollectionDefinition` would work.